manifests/server/service.pp

```diff
@@ -47,4 +47,18 @@
     File['mysql-config-file'] -> Service['mysqld']
   }
 
+  if $mysql::server::override_options and $mysql::server::override_options['mysqld'] and $mysql::server::override_options['mysqld']['socket'] {
+    $mysqlsocket = $mysql::server::override_options['mysqld']['socket']
+  } else {
+    $mysqlsocket = $options['mysqld']['socket']
+  }
+
+  exec { 'wait_for_mysql_socket_to_open':
+    command     => "ls ${mysqlsocket}",
+    tries       => '3',
+    try_sleep   => '10',
+    subscribe   => Service['mysqld'],
+    refreshonly => true,
```

Comment:
rather than refreshonly, you may wanna use:

```puppet
exec { 'wait_for_mysql_socket_to_open':
  command   => '/bin/true',
  tries     => '3'.
  try_sleep => '10',
  subscribe => Service['mysqld'],
  unless    => "/bin/test -S ${mysqlsocket}",
}
```